### DISCUSSION TITLE
How do we use current media breakpoint in the code?

### DISCUSSION THREAD

**Discussion by kachar** [heroui-inc/heroui#164] [Feedback]
:wave: 

I'm trying to achieve the following - container that is `direction: row` on desktop and `direction: column` on mobile

```tsx
const xs = useMediaQuery('xs')

return (
  <Container direction={xs ? 'column' : 'row'}>
    contents
  </Container>
)
```

I'm not able to find any documentation on using media queries and hooks together.

Great lib, kudos for the energy

**Comment by dartilesm** [comment]
I think you need to pass a number instead of string.
https://github.com/nextui-org/nextui/blob/b3ecba642eb907be57db06e2a27e394f816545c0/apps/docs/src/hooks/use-media-query.ts#L3

**Reply by kachar** [reply]
I never found the hook `useMediaQuery` to be documented, so I was guessing that there might be such a hook (coming from [Mui experience](https://mui.com/components/use-media-query/)).

Just realized that this hook is not exported member of `nextui`, but a simple `window.matchMedia` with hardcoded values, not based on the theme breakpoints.

https://github.com/nextui-org/nextui/blob/b3ecba642eb907be57db06e2a27e394f816545c0/apps/docs/src/hooks/use-media-query.ts#L29-L31

Are there plans to have this utility hook included in the package?

**Comment by jrgarciadev** [comment]
Hey @kachar @dartilesm since NextUI is based on Stitches and has built-in [media queries](https://nextui.org/docs/theme/media), you can simply add it in this way:

```jsx
 <Container css={{ 
    flexDirection: 'row',
    '@xsMax': {
      flexDirection: 'column'
    } 
 }}>
     // children ....
</Container>
```

> Note: if you prefer you can also use the shorthand `fd` instead of `flexDirection` here more info -> https://nextui.org/docs/theme/utilities#utilities

for further information, you can check out these pages:

1. [Stitches breakpoints](https://stitches.dev/docs/breakpoints)
2. [NextUI Built-in breakpoints](https://nextui.org/docs/theme/default-theme#breakpoints)
3. [Stitches responsive styles](https://stitches.dev/docs/responsive-styles)

Please try to avoid using the `useMediaQuery` to apply styles (unless absolutely necessary) due to it won't be included in the SSR styles

I hope this helps 🙌🏻

**Reply by kachar** [reply]
Thanks for the detailed reply, it was what I needed to learn about nextui principles.
